workers-docs/src/content/reference/apis/fetch.md

```diff
@@ -1,22 +1,40 @@
 ---
-title: fetch
+title: fetch()
 weight: 2
 ---
 
 ## Overview
 
-The [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) provides an interface for asyncronously fetching resources by providing a definition of a request and response. You will frequently find yourself interacting with request objects included as part of a [FetchEvent](/reference/apis/fetch-event), making your own requests using the global `fetch` method, and constructing your own responses.
-
-\*_Note: The Fetch API is only available inside of [the Request Context](/about/tips/request-context)._
+The [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) provides an interface for asyncronously fetching resources via HTTP requests inside of a Worker.
 
 ## Global
 
-The `fetch` method is implemented on the ServiceWorkerGlobalScope and matches the documentation [provided by MDN](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch)
+The `fetch` method is implemented on the ServiceWorkerGlobalScope. See [MDN documentation ](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch) for more information.
+
+_Note: Asynchronous tasks such as `fetch` are not executed at the top level in a Worker script and must be executed within a FetchEvent handler such as [`respondWith`](/reference/apis/fetch-event#methods). Learn more about [Request Contexts](/about/tips/request-context)._
+
+## Syntax
+
+`fetch(request:`[`Request`](/reference/apis/request)`| String, init?:`[`RequestInit`](/reference/apis/request)`) : Promise <`[`Response`](/reference/apis/response)`>`
+
+Where request is the request or the string of the URL.
 
-## Headers
+## Usage
 
-The Headers class matches the documentation [provided by MDN](https://developer.mozilla.org/en-US/docs/Web/API/Headers). If you expect Unicode values in your headers, URL or Base64 encode your header values before adding them to a Headers object.
+```js
+addEventListener('fetch', (event) => {
+  event.respondWith(eventHandler(event))
+})
+// note: don’t use fetch here
+async function eventHandler(event) {
+  // fetch available here
+  const resp = await fetch(event.request)
+  return resp
+}
+```
```

Comment:
```suggestion
```js
addEventListener('fetch', (event) => {
  // NOTE: don’t use fetch here, as we're not in an async scope yet
  event.respondWith(eventHandler(event))
})

async function eventHandler(event) {
  // fetch can be awaited here since `event.respondWith()` waits for the Promise it receives to settle
  const resp = await fetch(event.request)
  return resp
}
```
```